contracts/Rentals.sol

```diff
@@ -420,7 +435,7 @@ contract Rentals is NonceVerifiable, NativeMetaTransaction, IERC721Receiver {
         }
 
         // Only transfer the ERC721 to this contract if it doesn't already have it.
-        if (_ownerOf(address(asset), _tokenId) != address(this)) {
+        if (!extend && !reRent && _ownerOf(address(asset), _tokenId) != address(this)) {
```

Comment:
Why do we need to check for these variables? Is not ok only to check the ownership with `_ownerOf` as before?